jaeger_client/tracer.py

```diff
@@ -167,8 +179,9 @@ def start_span(self,
         if valid_references and (parent is None or not parent.has_trace):
             parent = valid_references[0].referenced_context
 
-        rpc_server = tags and \
-            tags.get(ext_tags.SPAN_KIND) == ext_tags.SPAN_KIND_RPC_SERVER
+        rpc_server = cast(
+            bool, tags and tags.get(ext_tags.SPAN_KIND) == ext_tags.SPAN_KIND_RPC_SERVER
+        )
```

Comment:
when I run `make lint` it works fine without this change. How are you testing typing?

Reply:
Did you also remove "bool" from the other line. I see then
```
jaeger_client/tracer.py:223: error: Argument "join" to "_emit_span_metrics" of "Tracer" has incompatible type "Union[Dict[Any, Any], None, Any]"; expected "Optional[bool]"
```

Reply:
```
$ make lint
flake8 jaeger_client crossdock tests
mypy jaeger_client
Success: no issues found in 36 source files
./scripts/check-license.sh

$ gd | cat
diff --git a/jaeger_client/tracer.py b/jaeger_client/tracer.py
index e4e8e00..e553f6e 100644
--- a/jaeger_client/tracer.py
+++ b/jaeger_client/tracer.py
@@ -22,6 +22,8 @@ import random
 import sys
 import time
 import opentracing
+from typing import Optional
+
 from opentracing import Format, UnsupportedFormatException
 from opentracing.ext import tags as ext_tags
 from opentracing.scope_managers import ThreadLocalScopeManager
@@ -282,7 +284,7 @@ class Tracer(opentracing.Tracer):
         self.sampler.close()
         return self.reporter.close()

-    def _emit_span_metrics(self, span, join=False):
+    def _emit_span_metrics(self, span: Span, join: Optional[bool] = False):
         if span.is_sampled():
             self.metrics.spans_started_sampled(1)
         else:
```

Reply:
I did `pip install mypy`
```
mypy-0.910
mypy_extensions-0.4.3
```

Reply:
So, of course if you remove the type annotation of the argument, mypy will not complain 😄

Reply:
I do have the type declared in the function. I also tried with my original suggestion, still successful:
```
        rpc_server: bool = tags and \
            tags.get(ext_tags.SPAN_KIND) == ext_tags.SPAN_KIND_RPC_SERVER
```

Reply:
This one should fail, because a cast is needed. Not sure, why it's successful on your side. On mine it fails as expected. 
Maybe a weird mypy cache issue or something else in your setup. Might also be related to the python version which you running.

However, making a cast is fine, so I see no risk here

Reply:
Yes, but the cast is ugly code. What errors are you getting with `rpc_server: bool =` ?

I'm running with Python 3.8.2

Reply:
```
jaeger_client/tracer.py:182: error: Incompatible types in assignment (expression has type "Union[Dict[Any, Any], None, Any]", variable has type "bool")
```

Reply:
Ok, found a way to write it easier